crates/rust-analyzer/src/handlers/request.rs

```diff
@@ -1995,14 +1996,24 @@ fn run_rustfmt(
             cmd
         }
         RustfmtConfig::CustomCommand { command, args } => {
-            let mut cmd = process::Command::new(command);
+            let cmd = PathBuf::from(&command);
+            let workspace = CargoTargetSpec::for_file(&snap, file_id)?;
+            let mut cmd = match workspace {
+                Some(spec) => {
+                    let cmd = spec.workspace_root.join(cmd);
+                    process::Command::new(cmd.as_os_str())
+                }
+                None => process::Command::new(cmd),
+            };
```

Comment:
This is treating stuff like `rustfmt` as relative paths still if I see this right? Iiirc from the discussion in that case we want to treat it as a plain command though (such that `PATH` lookup works)